src/main/kotlin/org/rust/lang/core/types/Extensions.kt

```diff
@@ -32,19 +31,19 @@ val RsTypeElement.lifetimeElidable: Boolean get() {
     return typeOwner !is RsFieldDecl && typeOwner !is RsTupleFieldDecl && typeOwner !is RsTypeAlias
 }
 
-val RsFunction.inference: RsInferenceResult
+val RsTypeInferenceRoot.inference: RsInferenceResult
     get() = CachedValuesManager.getCachedValue(this, {
         CachedValueProvider.Result.create(inferTypesIn(this), PsiModificationTracker.MODIFICATION_COUNT)
     })
 
 val PsiElement.inference: RsInferenceResult?
```

Comment:
Could `PsiElement.inference` and `RsTypeInferenceRoot.inference` have different names? They do slightly different things, and the case where `x`'s static type is `RsCompositteElement`, but its runtime type is `RsFunction` is murky.

Reply:
And can we make `PsiElement.inference` private? It looks overly general I think :)

Reply:
:thinking: Hmm, I think we can just unify behavior of these props. Just use `ancestorOrSelf` here. 
Anyway, the name `inference` looks really ugly.
I can't make it private because it is used in method/field/path reference impls